headers/augment_headers.py

```diff
@@ -214,6 +226,37 @@ def add_aliases(self) -> int:
                         symbol_declaration = []
         self.commit_header_file()
         return add_count
+    
+    def add_deprecated_macro(self):
+        # Adds the `DEPRECATED(message)` macro to the header file if it doesn't exist yet.
+        # This is a workaround for compatibility with clang-format
+        # (see https://stackoverflow.com/questions/76898417/make-clang-format-break-after-attribute)
+
+        macro = textwrap.dedent("""
+            #ifndef DEPRECATED
+            #define DEPRECATED(name) __attribute__((deprecated("Renamed to '" name "'")))
+            #endif
+        """)
+
+        with open(self.input_header_file(), "r") as f:
+            lines = f.readlines()
+
+        with open(self.output_header_file(), "w") as f:
+            in_guard = False
+            for line in lines:
+                if line.strip() == "#ifndef DEPRECATED":
+                    in_guard = True
+
+                if not in_guard:
+                    f.write(line)
+                
+                if line.strip() == "#endif" and in_guard:
```

Comment:
Let's put `in_guard` first so we don't need to do string operations unless we're actually within the guard.